lib/shared/src/models/sync.ts

```diff
@@ -485,6 +502,26 @@ export const maybeAdjustContextWindows = (models: ServerModel[]): ServerModel[]
             // of using a slightly smaller context window than what's available for those languages.
             maxInputTokens = Math.round(model.contextWindow.maxInputTokens * 0.85)
         }
+
+        // longContextWindow feature flag for testing the new context windows
+        if (longContextWindowFlag !== undefined) {
+            let maxOutputTokens = model.contextWindow.maxOutputTokens
+            if (
+                !longContextWindowFlag &&
+                (model.tier === ModelTag.Free || INPUT_TOKEN_FLAG_ON.includes(maxInputTokens))
+            ) {
+                maxInputTokens = INPUT_TOKEN_FLAG_OFF
+            }
+            if (isDotComUser && model.tier === ModelTag.Pro) {
```

Comment:
```suggestion
            if (isDotComUser && model.tier === ModelTag.Pro) {
```

This is going to be true to **ALL** DotComUser, because the same server side list is used by all DotComUser. E.g: 3.5 Sonnet is going to be a Free tier model for any users, and o1 is a Pro tier users no matter what the tier user is in.  That might be the confusion here